test/test_testing.py

```diff
@@ -4073,6 +4073,226 @@ def test_add_layer_merge(self, client: _TestingPebbleClient):
             == plan.to_yaml()
         )
 
+    def test_add_layer_checks_combine_override_replace(self, client: _TestingPebbleClient):
```

Comment:
This is quite a lot of YAML. Can we minimize the test cases to test just the features we're adding?

Reply:
Good point, removed.